lib/html_parser/namespacer/index.js

```diff
@@ -89,5 +89,71 @@ function namespace(htmlInput, prefix, postfix, cb) {
   cb(null, outputCache.join(''));
 }
 
+function stripNamespace(htmlInput, prefix, postfix, cb) {
+  var outputCache = [];
+
+  var omit = function(node) {
+    outputCache.push(node);
+  }.bind(this);
+
+  var namespacer = caja.makeSaxParser({
+    startdoc: omit,
+
+    startTag: function(tagNameLowerCase, attribs, outputCache) {
+      outputCache.push('<' + tagNameLowerCase); // output open tag and the tag's name
+
+      var hasClass = false;
+      var blacklisted = blacklister.verify(tagNameLowerCase);
+
+      if (!blacklisted) {
+        // parse attributes
+        for (var i = 0; i < attribs.length; i += 2) {
+          var key = attribs[i];
+          var value = attribs[i + 1];
+          // if parsing an "id" that has content
+          if (key === 'id' && value.indexOf(prefix) > -1) {
+            outputCache.push(' id="' + value.replace(prefix, '') + '"');
+          }
+
+          // if parsing an "class" that has content
+          else if (key === 'class' && value) {
+            hasClass = true;
+            value = value.replace(postfix, '').trim();
+            outputCache.push(' class="');
```

Comment:
I see `hasClass` is not used anywhere. Ideally if there are no class names left (they all have been removed). We should not add ` class=""` to the html output.